### PR TITLE
Override the error log callback for sqlite db

### PR DESCRIPTION
**Pull request by jinquan2019**
Things to consider before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [X] Are tests passing locally?
* [X] Are the files formatted correctly?
* [ ] Did you add unit tests?
* [ ] Did you check UI tests on the sample app? They are not executed on CI.
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description
Configure the DB at MSDBStorage load time to make sure not even the customer opened an SQLite DB

### CODE REVIEW

File: AppCenter/AppCenter/Internals/Storage/MSDBStorage.m [hunk 1]
```diff
@@ -15,8 +17,7 @@ - (instancetype)initWithSchema:(MSDBSchema *)schema version:(NSUInteger)version
     _dbFileURL = [MSUtility createFileAtPathComponent:filename withData:nil atomically:NO forceOverwrite:NO];
     _maxSizeInBytes = kMSDefaultDatabaseSizeInBytes;
```
Comment (Jamminroot) [inline AppCenter/AppCenter/Internals/Storage/MSDBStorage.m]:
```suggestion
```

File: AppCenter/AppCenter/Internals/Storage/MSDBStorage.m [hunk 2]
```diff
@@ -47,8 +48,7 @@ - (instancetype)initWithVersion:(NSUInteger)version filename:(NSString *)filenam
...
```
Comment (Jamminroot) [inline AppCenter/AppCenter/Internals/Storage/MSDBStorage.m]:
```suggestion
```

File: AppCenter/AppCenter/Internals/Storage/MSDBStorage.m [hunk 3]
```diff
@@ -317,6 +317,20 @@ - (void)customizeDatabase:(void *)__unused db {
...
+- (void)configSqliteDB {
+
+  // If it is custom SQLite library we need to turn on URI filename capability.
+  char *pData = NULL;
+  sqlite3_config(SQLITE_CONFIG_LOG, errorLogCallback, pData);
+  sqlite3_config(SQLITE_CONFIG_URI, 1);
+}
+
+void errorLogCallback(void *pArg, int iErrCode, const char *zMsg) {
+  (void)pArg;
```
Comment (MatkovIvan) [inline AppCenter/AppCenter/Internals/Storage/MSDBStorage.m]: Add logging here

File: AppCenter/AppCenter/Internals/Storage/MSDBStorage.m [hunk 4]
```diff
@@ -7,6 +7,8 @@
 #import "MSDBStoragePrivate.h"
 #import "MSUtility+File.h"
 
+void errorLogCallback(void *pArg, int iErrCode, const char *zMsg);
```
Comment (MatkovIvan) [inline AppCenter/AppCenter/Internals/Storage/MSDBStorage.m]: static

File: AppCenter/AppCenter/Internals/Storage/MSDBStorage.m [hunk 5]
```diff
@@ -317,6 +317,20 @@ - (void)customizeDatabase:(void *)__unused db {
 - (void)migrateDatabase:(void *)__unused db fromVersion:(NSUInteger)__unused version {
 }
 
+- (void)configSqliteDB {
+
+  // If it is custom SQLite library we need to turn on URI filename capability.
```
Comment (MatkovIvan) [inline AppCenter/AppCenter/Internals/Storage/MSDBStorage.m]: This comment is for a different line. Please move it.

File: AppCenter/AppCenter/Internals/Storage/MSDBStorage.m [hunk 6]
```diff
@@ -317,6 +317,20 @@ - (void)customizeDatabase:(void *)__unused db {
...
+  char *pData = NULL;
+  sqlite3_config(SQLITE_CONFIG_LOG, errorLogCallback, pData);
```
Comment (MatkovIvan) [inline AppCenter/AppCenter/Internals/Storage/MSDBStorage.m]:
```suggestion
  sqlite3_config(SQLITE_CONFIG_LOG, errorLogCallback, NULL);
```

File: AppCenter/AppCenter/Internals/Storage/MSDBStorage.m [hunk 7]
```diff
@@ -398,4 +394,11 @@ + (int)setMaxPageCount:(long)maxPageCount inOpenedDatabase:(void *)db {
   return [MSDBStorage executeNonSelectionQuery:statement inOpenedDatabase:db];
 }
 
++ (void)load {
```
Comment (jaelim-ms) [inline AppCenter/AppCenter/Internals/Storage/MSDBStorage.m]: This method isn't used.

Comment (jinquan2019) [reply AppCenter/AppCenter/Internals/Storage/MSDBStorage.m]: This method will be invoked before the class initializes

Comment (Jamminroot) [reply AppCenter/AppCenter/Internals/Storage/MSDBStorage.m]: I think he implies [this](https://developer.apple.com/documentation/objectivec/nsobject/1418815-load?language=objc). Looks good to me now when I remembered this one

Comment (clpolet) [reply AppCenter/AppCenter/Internals/Storage/MSDBStorage.m]: Yes the method is called by the runtime when the class is loaded. The SDK doesn't need to call it.

Comment (jaelim-ms) [reply AppCenter/AppCenter/Internals/Storage/MSDBStorage.m]: OK. Good to know. 😄

**Pull request merged**